### PR TITLE
Accuracy circle and obscuration cell for obsdetail map

### PR DESCRIPTION
**Pull request by angielt**
- Accuracy Circle and Obscuration Cell
- Details Tab Map Screen 

Part of the way toward #655, with the following exceptions:

- Dashed lines for accuracy circle and obscuration cell omitted since it's only available on iOS according to docs for react-native-maps(https://github.com/react-native-maps/react-native-maps/blob/master/docs/circle.md). 
- Waiting on designs for obscuration based on me vs taxon geoprivacy

### CODE REVIEW

**Review by kueda — CHANGES_REQUESTED** [review]
This is a partial review, but I'm guessing it's plenty to work on.

- [ ] You changed the database schema so you need to bump the schema version in src/realmModels/index.js, otherwise the app won't start if it has an existing database with a prior schema
- [ ] The map on the DETAILS tab of ObsDetails doesn't show any of the obscuration cell, positional accuracy circle, etc.
- [ ] https://www.inaturalist.org/observations/159598562 should have an obscuration cell, but instead I see a stemmed map marker and accuracy circle at the position of the *obscured* coordinates

File: src/realmModels/index.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ export default {
     User,
     Vote
   ],
-  schemaVersion: 39,
+  schemaVersion: 42,
```
Comment (kueda) [inline src/realmModels/index.js:29]: Why 42 and not 40?

**Review by kueda — CHANGES_REQUESTED** [review]
Another round of feedback! Holler if you have any questions. Here are some that didn't fit inline:

- [x] Copy Coordinates button closes DetailsMap when the toast fades away
- [x] Missing [geoprivacy indicator](https://www.figma.com/file/MChpvx4ZrKEVVwsWKt4lkI/iNaturalist-Mobile-UI-Design?type=design&node-id=1493-12165&mode=design&t=hjZ2bF9xKLohwFyA-4) below the map on ObsDetail

File: src/components/SharedComponents/Map.js [hunk 2]
```diff
@@ -34,10 +47,34 @@ const Map = ( {
...
+    <LocationIndicator width={25} height={32} />
+  );
+
+  const obscuredlLocationIndicator = () => (
+    // $FlowIgnore
+    <ObscuredLocationIndicator width={31} height={31} />
+  );
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const obscuredCoordinates = useMemo( () => returnRandomArbitraryCoordinates(), [] );
```
[the remark this answers is not in the document]
Comment (angielt) [reply src/components/SharedComponents/Map.js:76]: Really? I didn't realize that they were. I'll remove this then.

[the remark this answers is not in the document]
Comment (kueda) [reply src/components/SharedComponents/Map.js]: Yeah, the server decides what the client has access to. The client's responsibility is to *display* the location properly (e.g. with an obscuration cell or with an accuracy circle), but the client doesn't need to actually perform the obscuration itself.

**Review by kueda — CHANGES_REQUESTED** [review]
Getting closer! Here are a few more in addition to the inline feedback:

- [ ] The map marker is using the obscured coordinates even when the private coordinates are available. The map should show `privateLatitude` and `privateLongitude` if they are available. The symptom I'm seeing is that when I look at an observation of mine where geoprivacy is `obscured`, the map marker looks unobscured (stemmed map marker), but the location is wrong (should be in my back yard, is in another town instead).
- [ ] When I tap the share button on the DetailsMap, it opens Apple Maps on iOS and Google Maps on Android, but I don't see a map marker on either. Is that expected?
- [ ] Resolve conflicts with main

File: src/components/SharedComponents/ObservationLocation.js [hunk 3]
```diff
@@ -35,18 +40,37 @@ const ObservationLocation = ( {
...
-        numberOfLines={1}
-        ellipsizeMode="tail"
-      >
-        {geoprivacy}
-      </Body4>
-    </View>
-  );
+  const displayGeoprivacy = ( ) => {
+    let displayPrivacy = geoprivacy;
+    if ( obscured ) {
```
[the remark this answers is not in the document]
Comment (angielt) [reply src/components/SharedComponents/ObservationLocation.js:45]: I see. So coordinates obscured takes into account the taxon geoprivacy. For several of my observations and []((https://www.inaturalist.org/observations/159598562)) i'm seeing geoprivacy as null.  A bit confused on what to do in this case.

[the remark this answers is not in the document]
Comment (kueda) [reply src/components/SharedComponents/ObservationLocation.js:45]: `observation.geopricacy == null` should be treated the same as `observation.geoprivacy == "open"`.

**Review by kueda — APPROVED** [review]

**Pull request merged**